### PR TITLE
Moved all camera 'setting' option code to single source

### PR DESCRIPTION
**Pull request by JamesH65**
All the cam apps were duplicating the same code for outputing camera
settings (e.g. exposure), so have moved it to the CamControl file
so we now only have one set.

Also fixed up a couple of missing bad error returns that were not being
translated correctly from the MMAL error to ints.

### CODE REVIEW

File: host_applications/linux/apps/raspicam/RaspiStill.c [hunk 1]
```diff
@@ -216,7 +215,6 @@ static void store_exif_tag(RASPISTILL_STATE *state, const char *exif_tag);
 #define CommandSignal       16
 #define CommandGL           17
 #define CommandGLCapture    18
-#define CommandSettings     19
```
Comment (6by9) [inline host_applications/linux/apps/raspicam/RaspiStill.c]: Does it cause a problem having a gap in the numbers? Do we rewrite this as an enum definition?

Comment (JamesH65) [reply host_applications/linux/apps/raspicam/RaspiStill.c]: No problem with gaps, but I'll change to enums in a different PR anyway. unless I remember why I did that way originally...

**Comment by JamesH65** [comment]
I did wonder about calling it 'default_camera_control_callback`, or having a small shim, perhaps make the name change anyway, even if I don't add a shim, which just seems like adding code for codes sake.

**Comment by 6by9** [comment]
To shim, or not to shim, that is the question!
I'm happy not to have one if you make it ```default_camera_control_callback```, as at least it is then obvious what is going on.

**Comment by JamesH65** [comment]
OK, changed name - also fixed some build warnings as I was there.

**Pull request merged**